examples/amm/src/contract.rs

```diff
@@ -0,0 +1,112 @@
+#![cfg_attr(target_arch = "wasm32", no_main)]
+
+mod state;
+
+use self::state::Amm;
+use amm::{get_pool_balance, update_pool_balance, AMMError};
+use async_trait::async_trait;
+use linera_sdk::{
+    base::{SessionId, WithContractAbi},
+    ApplicationCallResult, CalleeContext, Contract, ExecutionResult, MessageContext,
+    OperationContext, SessionCallResult, ViewStateStorage,
+};
+
+linera_sdk::contract!(Amm);
+
+impl WithContractAbi for Amm {
+    type Abi = amm::AMMAbi;
+}
+
+#[async_trait]
+impl Contract for Amm {
+    type Error = AMMError;
+    type Storage = ViewStateStorage<Self>;
+
+    async fn initialize(
+        &mut self,
+        _context: &OperationContext,
+        _argument: (),
+    ) -> Result<ExecutionResult<Self::Message>, AMMError> {
+        Ok(ExecutionResult::default())
+    }
+
+    async fn execute_operation(
+        &mut self,
+        _context: &OperationContext,
+        operation: Self::Operation,
+    ) -> Result<ExecutionResult<Self::Message>, AMMError> {
+        match operation {
+            Self::Operation::Swap {
+                input_token,
+                output_token,
+                amount,
+            } => {
```

Comment:
Will this be automated in the future?
I imagine we'll have bids placed using cross-chain messages, and then the AMM automatically fills them?

Reply:
We could! I implemented the simplest thing I could think of, but happy to evolve it to have more functionality according to the feedback from you guys